File: TaskWorkFlowManagement.Api/Models/TaskItem.cs
namespace TaskWorkFlowManagement.Api.Models;

public class TaskItem
{
    public const int TitleMaxLength = 200;

    public Guid Id { get; set; } = Guid.NewGuid();

    public string Title { get; set; } = string.Empty;

    public string? Description { get; set; }

    public TaskItemStatus Status { get; set; } = TaskItemStatus.ToDo;

    public DateTime CreatedAtUtc { get; set; }
}

File: TaskWorkflowManagement.Web/src/app/models/task-item.ts
import type { TaskItemStatus } from './task-item-status';

export const TASK_ITEM_TITLE_MAX_LENGTH = 200;

export interface TaskItem {
  id: string;
  title: string;
  description: string | null;
  status: TaskItemStatus;
  createdAtUtc: string;
}

export interface CreateTaskItemRequest {
  title: string;
  description: string | null;
}

export interface UpdateTaskItemRequest {
  title: string;
  description: string | null;
}

export interface UpdateTaskItemStatusRequest {
  status: TaskItemStatus;
}
